### PR TITLE
fix: support usernames with multiple dashes

### PR DESCRIPTION
**Pull request by brandonroberts**
**What**:

Fixes a bug where a username with multiple dashes is not formatted correctly and the username is not found 

**Why**:

https://github.com/analogjs/analog/pull/362#issuecomment-1529114039

image

**How**:

Replaces all occurrences of a dash in the username. Removes all occurrences of the replacement characters

**Checklist**:

- [ ] Documentation
- [x] Ready to be merged 
- [ ] Added myself to contributors table. 
[Bot Usage](https://allcontributors.org/docs/en/bot/installation#4-update-your-contributing-documentation)

### CODE REVIEW

File: test/unit/parse-comment.test.js [hunk 1]
```diff
@@ -62,6 +62,17 @@ describe('parseComment', () => {
...
+  test('Basic intent to add - username with multiple dashes', () => {
+    expect(
+        parseComment(`@${testBotName} please add rishi-raj-jain for doc`),
+    ).toEqual({
+        action: 'add',
+        contributors: {
+            "rishi-raj-jain": ['doc'],
+        },
+    })
+})
```
Comment (gr2m) [inline test/unit/parse-comment.test.js:74]: The test is not failing. Looks like the existing code is working, even without your change?

Comment (gr2m) [reply test/unit/parse-comment.test.js:74]: here is a test, seems like the username is parsed okay?

https://github.com/gr2m/sandbox/issues/252#issuecomment-1539284970

Comment (brandonroberts) [reply test/unit/parse-comment.test.js:74]: Hmm, that's where I traced it down to parsing the username. It shows a valid username not being parsed here https://github.com/analogjs/analog/pull/362#issuecomment-1529114039

Comment (gr2m) [reply test/unit/parse-comment.test.js:74]: hm okay I'll do some more testing

Comment (gr2m) [reply test/unit/parse-comment.test.js:74]: okay I was able to reproduce the problem: https://github.com/gr2m/sandbox/issues/252#issuecomment-1546400131.

As I said the test is not failing the implementation of your fix though. Could you please have a look? I'll meanwhile deploy your changes to staging and test that

Comment (gr2m) [reply test/unit/parse-comment.test.js:74]: confirmed your change is working: https://github.com/gr2m/sandbox/issues/252#issuecomment-1546408394. So all it's left is to make your test fail before the fix.

Comment (brandonroberts) [reply test/unit/parse-comment.test.js:74]: Sure, thanks

**Review by gr2m — APPROVED** [review]

**Comment by gr2m** [comment]
@all-contributors add @brandonroberts for code and test

**Pull request merged**